vpr/src/pack/greedy_seed_selector.cpp

```diff
@@ -0,0 +1,237 @@
+/**
+ * @file
+ * @author  Alex Singer
+ * @date    November 2024
+ * @brief   The definitions of the Greedy Seed Selector class.
+ */
+
+#include "greedy_seed_selector.h"
+
+#include <algorithm>
+#include "atom_netlist.h"
+#include "cluster_legalizer.h"
+#include "cluster_util.h"
+#include "echo_files.h"
+#include "prepack.h"
+#include "vpr_error.h"
+#include "vpr_types.h"
+#include "vtr_assert.h"
+#include "vtr_math.h"
+#include "vtr_vector.h"
+
+/**
+ * @brief Helper method that computes the seed gain of the give atom block.
+ *
+ * The seed_type variable selects which algorithm to use to compute the seed
+ * gain.
+ */
+static inline float get_seed_gain(AtomBlockId blk_id,
+                                  const AtomNetlist& atom_netlist,
+                                  const Prepacker& prepacker,
+                                  const e_cluster_seed seed_type,
+                                  const t_molecule_stats& max_molecule_stats,
+                                  const vtr::vector<AtomBlockId, float>& atom_criticality) {
+    switch (seed_type) {
+        // By criticality
```

Comment:
. Intuition: starting a cluster with primitives that have timing-critical connections may help timing.